dmriprep/config/__init__.py

```diff
@@ -1,6 +1,536 @@
+    spaces = None
+    """Keeps the :py:class:`~niworkflows.utils.spaces.SpatialReferences`
+    instance keeping standard and nonstandard spaces."""
+    use_syn = None
+    """Run *fieldmap-less* susceptibility-derived distortions estimation
+    in the absence of any alternatives."""
+
+
+class loggers:
+    """Keep loggers easily accessible (see :py:func:`init`)."""
+
+    _fmt = "%(asctime)s,%(msecs)d %(name)-2s " "%(levelname)-2s:\n\t %(message)s"
+    _datefmt = "%y%m%d-%H:%M:%S"
+
+    default = logging.getLogger()
+    """The root logger."""
+    cli = logging.getLogger('cli')
+    """Command-line interface logging."""
+    workflow = nlogging.getLogger('nipype.workflow')
+    """NiPype's workflow logger."""
+    interface = nlogging.getLogger('nipype.interface')
+    """NiPype's interface logger."""
+    utils = nlogging.getLogger('nipype.utils')
+    """NiPype's utils logger."""
+
+    @classmethod
+    def init(cls):
+        """
+        Set the log level, initialize all loggers into :py:class:`loggers`.
+
+            * Add new logger levels (25: IMPORTANT, and 15: VERBOSE).
+            * Add a new sub-logger (``cli``).
+            * Logger configuration.
+
+        """
+        from nipype import config as ncfg
+        _handler = logging.StreamHandler(stream=sys.stdout)
+        _handler.setFormatter(
+            logging.Formatter(fmt=cls._fmt, datefmt=cls._datefmt)
+        )
+        cls.cli.addHandler(_handler)
+        cls.default.setLevel(execution.log_level)
+        cls.cli.setLevel(execution.log_level)
+        cls.interface.setLevel(execution.log_level)
+        cls.workflow.setLevel(execution.log_level)
+        cls.utils.setLevel(execution.log_level)
+        ncfg.update_config({
+            'logging': {
+                'log_directory': str(execution.log_dir),
+                'log_to_file': True
+            },
+        })
+
+
+def from_dict(settings):
+    """Read settings from a flat dictionary."""
+    nipype.load(settings)
+    execution.load(settings)
+    workflow.load(settings)
+    loggers.init()
+
+
+def load(filename):
+    """Load settings from file."""
+    from toml import loads
+    filename = Path(filename)
+    settings = loads(filename.read_text())
+    for sectionname, configs in settings.items():
+        if sectionname != 'environment':
+            section = getattr(sys.modules[__name__], sectionname)
+            section.load(configs)
+    init_spaces()
+
+
+def get(flat=False):
+    """Get config as a dict."""
+    settings = {
+        'environment': environment.get(),
+        'execution': execution.get(),
+        'workflow': workflow.get(),
+        'nipype': nipype.get(),
+    }
+    if not flat:
+        return settings
+
+    return {'.'.join((section, k)): v
+            for section, configs in settings.items()
+            for k, v in configs.items()}
+
+
+def dumps(flat=False):
+    """Format config into toml."""
+    from toml import dumps
+    return dumps(get(flat=flat))
```

Comment:
It's confusing (to me) that the function is called `dumps`, and it invokes another function also called `dumps`. Maybe that's not too important.